### DISCUSSION TITLE
API Endpoint Issue

### DISCUSSION THREAD

**Discussion by tarilen** [simple-login/app#2099] [Self-hosting question]
I'm writing a script to register a list of users in bulk. When calling the API endpoint to register it says the user has to check their email for activation. 

Because I'm self-hosting through AWS I'm not going to get that activation email, but, I'm not seeing that request generate an activation code attached to the user in the database. 

When I register through the WebGUI it DOES generate the activation code even though I never get the email.

Attempting to hit the API endpoint to reactivate also does not send an email or generate an activation code in the database.

Is there another way I should go about this?

**Comment by nguyenkims** [comment]
We can do that directly in Python without calling an API endpoint.
For example this will create a user that is activated `user = User.create(email=email, activated=True, commit=True)`

**Reply by tarilen** [reply]
Thanks for the reply. I was hoping to resolve this through an API call to keep things consistent. It would seem that something with the user registration flow is different between the GUI and API calls which could explain why one is generating an activation code and one is not.

Is it safe to assume there is a bug or unintended workflow happening with registering via the API?

**Reply by nguyenkims** [reply]
The API is intended to be used by SL apps and users are expected to verify their email upon signup. So this is actually by design and not a bug.

**Comment by tarilen** [comment]
Thanks for the answer. I've solved with a workaround to not having an activation code or a verification email by setting it as active in the database. This way I can create users in bulk and not have to worry about an email activation link or code, in my scenario, where emails generated by SL aren't leaving the server.
